src/libs/Network/processRequest.js

```diff
@@ -11,11 +32,43 @@ import * as NetworkEvents from './NetworkEvents';
  * @returns {Promise}
  */
 export default function processRequest(request) {
+    const persisted = lodashGet(request, 'data.persist', false);
     const finalParameters = enhanceParameters(request.command, request.data);
 
     // When the request goes past a certain amount of time we trigger a re-check of the connection
     const cancelRequestTimeoutTimer = NetworkEvents.startRecheckTimeoutTimer();
-    NetworkEvents.triggerRequestMade(request, finalParameters);
+    logRequestDetails(request, finalParameters);
     return HttpUtils.xhr(request.command, finalParameters, request.type, request.shouldUseSecure)
+        .then((response) => {
+            if (persisted) {
+                PersistedRequests.remove(request);
+            }
+            NetworkEvents.triggerResponse(request, response);
+            return response;
+        })
+        .catch((error) => {
+            if (error.message === CONST.ERROR.FAILED_TO_FETCH) {
+                // Throw when we get a "Failed to fetch" error so we can retry. Very common if a user is offline or experiencing an unlikely scenario like
+                // incorrect url, bad cors headers returned by the server, DNS lookup failure etc.
+                throw error;
+            }
+
+            // Cancelled requests are normal and can happen when a user logs out. No extra handling is needed here besides
+            // remove the request from the PersistedRequests if the request exists.
+            if (error.name === CONST.ERROR.REQUEST_CANCELLED) {
+                NetworkEvents.getLogger().info('[Network] Request canceled', false, request);
+            } else {
+                // If we get any error that is not "Failed to fetch" create GitHub issue so we can handle it. These requests will not be retried.
+                NetworkEvents.getLogger().alert(`${CONST.ERROR.ENSURE_BUGBOT} unknown error caught while processing request`, {
```

Comment:
NAB: I think we should throw an error here, no? because if the promise is chained, the catch method of the upper promise won't be called (i.e. if we're updating the UI or displaying an error in the catch method, that won't happen)

Reply:
Something that I forgot to mention is that this comment is related to the previous one (if the retry logic was moved to processRequest function), otherwise seems this will cause an infinite loop. 

Given the reason that moving the retry is not a good idea, I think throwing an error is a no-go here.